### ISSUE TITLE
half precision not supported by CPU?

### ISSUE THREAD

**Issue by wolfshow** [zhihu/cuBERT#8]
cannot run the CPU code now.

**Comment by levyfan** [comment]
Yes, half precision is not supported by CPU. MKL does not have such half precision BLAS methods.

**Comment by wolfshow** [comment]
> Half precision is not supported by CPU. MKL does not have such half precision BLAS methods.

how to modify the code to be used with CPU?

**Comment by levyfan** [comment]
> > Half precision is not supported by CPU. MKL does not have such half precision BLAS methods.
> 
> how to modify the code to be used with CPU?

Could you clarify your question? I am not sure what you are asking.

For running on CPU, only single precision (float) is supported. The reason of no half precision is lying on Intel's MKL. They do not support half precision computing. There is nothing we can do now.

**Comment by wolfshow** [comment]
> For running on CPU, only single precision (float) is supported. The reason of no half precision is lying on Intel's MKL. They do not support half precision computing. There is nothing we can do now.

When I compile the current code base and run the ”cuBERT_benchmark“, I got the error "half precision not supported by CPU". My question is, how to modify the code and make it runnable on CPU?

**Comment by levyfan** [comment]
You can simply change https://github.com/zhihu/cuBERT/blob/8ab038415bda11c1cd715db2bc618a430d6093c1/benchmark/benchmark_cu.cpp#L8
to `typedef float Dtype;`

and change https://github.com/zhihu/cuBERT/blob/8ab038415bda11c1cd715db2bc618a430d6093c1/benchmark/benchmark_cu.cpp#L11
to `cuBERT_ComputeType compute_type = cuBERT_COMPUTE_FLOAT;`

**Comment by levyfan** [comment]
If you would like to run your own model, you may also need to give https://github.com/zhihu/cuBERT/blob/8ab038415bda11c1cd715db2bc618a430d6093c1/benchmark/benchmark_cu.cpp#L39
more memory space and change `output_type` from cuBERT_LOGITS to your own model.

**Comment by wolfshow** [comment]
> If you would like to run your own model, you may also need to give
> 
> [cuBERT/benchmark/benchmark_cu.cpp](https://github.com/zhihu/cuBERT/blob/8ab038415bda11c1cd715db2bc618a430d6093c1/benchmark/benchmark_cu.cpp#L39)
> 
> Line 39 in [8ab0384](/zhihu/cuBERT/commit/8ab038415bda11c1cd715db2bc618a430d6093c1)
> 
>  Dtype logits[batch_size]; 
> 
> more memory space and change `output_type` from cuBERT_LOGITS to your own model.

Thanks a lot!
